site/app/controllers/forum/ForumController.php

```diff
@@ -916,6 +916,18 @@ public function getThreads($page_number = null) {
      * @Route("/courses/{_semester}/{_course}/forum/threads/single", methods={"POST"})
      */
     public function getSingleThread() {
+        // Checks if thread id is null. If so, render "fail" json response case informing that thread id is null.
+        if ($_POST['thread_id'] == null) {
```

Comment:
I've been told that using === is always better.

Reply:
While testing, it looks like if I used `===` instead of `==`, the null value would get caught when when the function checks if the thread id is an integer, so I think it might be better to use `==` in this case.

Reply:
I also added a check to see if the thread id is unset using `isset`.  Interestingly, `isset` did not catch either of the aforementioned thread ids used when testing for null.

Reply:
You could look into using the `empty` function

Reply:
I looked into `empty`, and it handles of both `$_POST['thread_id']) == null` and  `!isset($_POST['thread_id']))`.  I'll change it to use `empty`.  Thank you.